File: VB/SnapExampleEF/BusinessObjects/DocumentData.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports SnapModule.Base
Imports DevExpress.Persistent.Base
Imports System.Collections
Imports System.ComponentModel

Namespace SnapExampleEF.BusinessObjects
    <NavigationItem("Reports"), ImageName("BO_Report"), System.ComponentModel.DisplayName("Document")> _
    Public Class DocumentData
        Implements IDocumentData

        Public Sub New()
            DataSources = New List(Of DocumentDataSource)()
        End Sub

        Private privateID As Int32
        <Browsable(False)> _
        Public Property ID() As Int32
            Get
                Return privateID
            End Get
            Protected Set(ByVal value As Int32)
                privateID = value
            End Set
        End Property

        Public Property DocumentName() As String Implements IDocumentData.DocumentName

        <Browsable(False)> _
        Public Property Content() As Byte() Implements IDocumentData.Content

        Public Overridable Property DataSources() As IList(Of DocumentDataSource)

        <DataSourceProperty("DataSources")> _
        Public Overridable Property InplaceDocumentDataSource() As DocumentDataSource

        Private ReadOnly Property IDocumentData_DataSources() As IList Implements IDocumentData.DataSources
            Get
                Return DirectCast(DataSources, IList)
            End Get
        End Property

        Private ReadOnly Property IDocumentData_InplaceDocumentDataSource() As IDocumentDataSource Implements IDocumentData.InplaceDocumentDataSource
            Get
                Return InplaceDocumentDataSource
            End Get
        End Property
    End Class
End Namespace

File: CS/SnapExampleEF/BusinessObjects/DocumentData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using SnapModule.Base;
using DevExpress.Persistent.Base;
using System.Collections;
using System.ComponentModel;

namespace SnapExampleEF.BusinessObjects {
    [NavigationItem("Reports")]
    [ImageName("BO_Report"), System.ComponentModel.DisplayName("Document")]
    public class DocumentData : IDocumentData {
        public DocumentData() {
            DataSources = new List<DocumentDataSource>();
		}

        [Browsable(false)]
        public Int32 ID { get; protected set; }

        public string DocumentName { get; set; }

        [Browsable(false)]
        public byte[] Content { get; set; }

        public virtual IList<DocumentDataSource> DataSources { get; set; }

        [DataSourceProperty("DataSources")]
        public virtual DocumentDataSource InplaceDocumentDataSource { get; set; }

        IList IDocumentData.DataSources {
            get {
                return (IList)DataSources;
            }
        }

        IDocumentDataSource IDocumentData.InplaceDocumentDataSource {
            get {
                return InplaceDocumentDataSource;
            }
        }
    }
}
